### PR TITLE
Add EC_KEY_get0_engine()

### LINKED ISSUE #1779: need a way to get engine filed of EVP_PKEY
need a way to get engine filed of EVP_PKEY, just like EVP_PKEY_get_engine(), version 1.1.x or master?

https://github.com/openssl/openssl/blob/6f0ac0e2f27d9240516edb9a23b7863e7ad02898/crypto/include/internal/evp_int.h

```c
struct evp_pkey_ctx_st {
    /* Method associated with this operation */
    const EVP_PKEY_METHOD *pmeth;
    /* Engine that implements this method or NULL if builtin */
    ENGINE *engine;
    /* Key: may be NULL */
    EVP_PKEY *pkey;
    /* Peer key for key agreement, may be NULL */
    EVP_PKEY *peerkey;
    /* Actual operation */
    int operation;
    /* Algorithm specific data */
    void *data;
    /* Application specific data */
    void *app_data;
    /* Keygen callback */
    EVP_PKEY_gen_cb *pkey_gencb;
    /* implementation specific keygen data */
    int *keygen_info;
    int keygen_info_count;
} /* EVP_PKEY_CTX */ ;
```

### PR DESCRIPTION
**Pull request by levitte**
Just as for DH, DSA and RSA, this gives the engine associated with the
key.

-----

Partially fixes #1779

### CODE REVIEW

**Review by mattcaswell — CHANGES_REQUESTED** [review]
Documentation?

Is this for backport to 1.1.0 (missing accessor)? Although, EC_KEY has always been opaque, so perhaps it doesn't qualify???

**Comment by levitte** [comment]
Because EC_KEY has always been opaque, I consider this a new feature.  Some might argue that it's an omission that should have been added ages ago, but there you go...

Point about docs, I'll get right to it.  And a note in CHANGES.

**Comment by levitte** [comment]
There, documentation added

File: doc/man3/EC_KEY_new.pod [hunk 1]
```diff
@@ -78,6 +80,9 @@ EC_KEY_dup() creates a new EC_KEY object and copies B<ec_key> into it.
 EC_KEY_up_ref() increments the reference count associated with the EC_KEY
 object.
 
+EC_KEY_get0_engine() returns a handle to the ENGINE that has been set for
+this RSA object.
```
Comment (mattcaswell) [inline doc/man3/EC_KEY_new.pod]: RSA? copy&paste error!

**Review by mattcaswell — APPROVED** [review]
Approved subject to the typo being fixed.

**Comment by levitte** [comment]
Merged.  d1da335c5514f9ba8b5115faab208a8046a5bc56

**Pull request closed without merging**